src/java.base/share/classes/java/lang/String.java

```diff
@@ -4453,12 +4453,34 @@ public String repeat(int count) {
         final int limit = len * count;
         final byte[] multiple = new byte[limit];
         System.arraycopy(value, 0, multiple, 0, len);
-        int copied = len;
+        repeatCopyRest(multiple, 0, limit, len);
+        return new String(multiple, coder);
+    }
+
+    /**
+     * Used to perform copying after the initial insertion. Copying is optimized
```

Comment:
It may be worth reinforceing that bytes are being copied, not characters and is used in both IsLatin1 and UTF16 cases with appropriate arguments.